contracts/hot-cold-aa.sol

```diff
@@ -140,72 +153,88 @@ contract HotColdMultiSig is IAccount, IERC1271 {
+        require(magic == ACCOUNT_VALIDATION_SUCCESS_MAGIC, "NOT VALIDATED");
 
-            magic = EIP1271_SUCCESS_RETURN_VALUE;
+        _executeTransaction(_transaction);
+    }
 
-            if (_signature.length != 130) {
-                // Signature is invalid, but we need to proceed with the signature verification as usual
-                // in order for the fee estimation to work correctly
-                _signature = new bytes(130);
+    function isValidSignature(bytes32 _hash, bytes memory _signature) public view override returns (bytes4 magic) {
+        // TO BE IMPLEMENTED
 
-                // Making sure that the signature looks like a valid ECDSA signature and we're not rejected
-                // rightaway while skipping the main verification process
-                _signature[64] = bytes1(uint8(27));
-                _signature[129] = bytes1(uint8(27));
-            }
+        magic = EIP1271_SUCCESS_RETURN_VALUE;
+        // COMMENT: Provide data on why we're using this format.
 
-            (bytes memory signature1, bytes memory signature2) = extractECDSASignature(_signature);
+        // we def need to verify -> signature corresponds to one of the wallets -> if we get a random sig, the wallet should not accept it
 
-            if(!checkValidEDCSASignatureFormat(signature1) || !checkValidECDSASignatureFormat(signature2)) {
-                magic = bytes4(0);
-            }
+        /*
+            Signature Validation Steps
+            1. Check if the length of the received signature is correct, equal to 66
+            2. Check if the signatures are in the correct format using the helper function 
+                `checkValidEDCSASignatureFormat`
+            3. We use EDCSASignatureFormat helper function, to check if the signature is in the correct format and is non-malleable.
+            4. 
+        
+         */
+        if (_signature.length != 66) {
+            // Signature is invalid, but we need to proceed with the signature verification as usual
+            // in order for the fee estimation to work correctly
+            _signature = new bytes(66);
+
+            // Making sure that the signature looks like a valid ECDSA signature and we're not rejected
+            // rightaway while skipping the main verification process
+            // TODO: since we moved forward the signature by one, change this as well
+            _signature[65] = bytes1(uint8(27));
+        }
 
-            address recoveredAddr1 = ECDSA.recover(_hash, signature1);
-            address recoveredAddr2 = EDCSA.recover(_hash, signature2);
+        if (!checkValidECDSASignatureFormat(_signature)) {
+            magic = bytes4(0);
+        }
 
-            // Note, that we should abstrain from using the required here in order to allow for fee estimation to work
+        address recoveredAddr = ECDSA.recover(_hash, _signature); // by this one we will know who signed the tx
```

Comment:
You should to take a slice from signature (take bytes array starting from index 1 to the end of signature). Because otherwise you will pass 66 bytes with the first byte is not expected to be a part of the ecdsa signature format
```suggestion
        address recoveredAddr = ECDSA.recover(_hash, _signature[1:]); // by this one we will know who signed the tx
```